### ISSUE TITLE
testrig crash when topology contains invalid hosts

### ISSUE THREAD

**Issue by virtuald** [batfish/batfish#209]
If we add a `topology.net` to the example cited in #207 that looks like so:

```
BATFISH_TOPOLOGY
'random':'e0', 'stuff':'eth0'
```

Then batfish will crash when I run a query with the following stack trace:

```
Caused by: java.lang.NullPointerException
   at org.batfish.bdp.BdpDataPlane.lambda$1(BdpDataPlane.java:164)
   at java.util.TreeMap.forEach(TreeMap.java:1005)
   at org.batfish.bdp.BdpDataPlane.lambda$0(BdpDataPlane.java:64)
   at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
   at java.util.TreeMap$ValueSpliterator.forEachRemaining(TreeMap.java:2897)
   at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
   at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
   at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
   at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
   at java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:401)
   at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:734)
   at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:160)
   at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:174)
   at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
   at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
   at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:583)
   at org.batfish.bdp.BdpDataPlane.getFibs(BdpDataPlane.java:58)
   at org.batfish.z3.Synthesizer.<init>(Synthesizer.java:1010)
   at org.batfish.main.Batfish.synthesizeDataPlane(Batfish.java:4432)
   at org.batfish.main.Batfish.standard(Batfish.java:4334)
   at org.batfish.question.ReachabilityQuestionPlugin$ReachabilityAnswerer.standard(ReachabilityQuestionPlugin.java:83)
   at org.batfish.question.ReachabilityQuestionPlugin$ReachabilityAnswerer.answer(ReachabilityQuestionPlugin.java:38)
   at org.batfish.main.Batfish.answer(Batfish.java:533)
```

It seems like it either should not crash (eg, check to see if edges is null at BdpDataPlane.java:159), or it should validate the initial testrig configuration to make sure it's sane.

**Comment by hverma-upenn** [comment]
@virtuald , Do you remember the query which you ran?

**Comment by virtuald** [comment]
A get reachability query.

**Comment by hverma-upenn** [comment]
Resolved by PR #318
